lib/controllers/cats.js

```diff
@@ -0,0 +1,34 @@
+const { Router } = require('express');
+const Cat = require('../models/Cat');
+
+module.exports = Router()
+  // for api/v1/cats
+  .post('/', async (req, res) => {
+    const cat = await Cat.insert(req.body);
+    res.send(cat);
+  })
+
+  .get('/', async (req, res) => {
+    const cats = await Cat.findALL();
+    res.send(cats);
+  })
+  // api/v1/cats/:id
+  .get('/:id', async (req, res, next) => {
+    try {
+      const cat = await Cat.findById(req.params.id);
+      res.send(cat);
+    } catch (error) {
+      error.status = 404;
```

Comment:
We might consider only setting the status to 404 in the case of a cat not being found